core/src/plugins/filed/python/percona-xtrabackup/bareos-fd-percona-xtrabackup.py

```diff
@@ -522,5 +564,56 @@ def restore_object_data(self, ROP):
             )
         return bRC_OK
 
+    def end_dumper(self):
+        """
+        Wait for bareos_xtrabackup_dumper to terminate
+        """
+        bareos_xtrabackup_dumper_returncode = None
+        # Wait timeout before sending TERM to the process
+        timeout = 30
+        start_time = int(time.time())
+        sent_sigterm = False
+        while self.stream.poll() is None:
+            if int(time.time()) - start_time > timeout:
+                DebugMessage(
+                    100,
+                    "Timeout wait for bareos_xtrabackup_dumper PID %s to terminate\n"
+                    % (self.stream.pid),
+                )
+                if not sent_sigterm:
+                    DebugMessage(
+                        100,
+                        "sending SIGTERM to bareos_xtrabackup_dumper PID %s\n"
+                        % (self.stream.pid),
+                    )
+                    os.kill(self.stream.pid, signal.SIGTERM)
+                    sent_sigterm = True
+                    # Wait timeout after sending TERM to the process
+                    timeout = 30
+                    start_time = int(time.time())
+                    continue
+                else:
+                    DebugMessage(
+                        100,
+                        "Giving up to wait for bareos_xtrabackup_dumper PID %s to terminate\n"
+                        % (self.stream.pid),
+                    )
+                    break
```

Comment:
```suggestion
                    # Set return code to 9 because we've tried to terminate subprocess earlier
                    self.stream.returncode = 9
                    break
```
I think this is missing here, otherwise it could happen that end_dumper() would return None,
and then the line after calling `self.end_dumper()` would fail because it compares it using `!= 0`